packages/flutter/lib/src/widgets/focus_traversal.dart

```diff
@@ -0,0 +1,783 @@
+///
+/// By default, traverses in widget order using
+/// [WidgetOrderFocusTraversalPolicy].
+///
+/// See also:
+///
+///   * [FocusNode], for a description of the focus system.
+///   * [WidgetOrderFocusTraversalPolicy], a policy that relies on the widget
+///     creation order to describe the order of traversal.
+///   * [ReadingOrderTraversalPolicy], a policy that describes the order as the
+///     natural "reading order" for the current [Directionality].
+///   * [DirectionalFocusTraversalPolicyMixin] a mixin class that implements
+///     focus traversal in a direction.
+class DefaultFocusTraversal extends InheritedWidget {
+  /// Creates a FocusTraversal object.
+  ///
+  /// The [policy] and [child] arguments must not be null.
+  const DefaultFocusTraversal({
+    Key key,
+    this.policy = const WidgetOrderFocusTraversalPolicy(),
+    @required Widget child,
+  })  : assert(policy != null),
+        super(key: key, child: child);
+
+  /// The policy used to move the focus from one focus node to another.
+  ///
+  /// By default, traverses in widget order using
+  /// [WidgetOrderFocusTraversalPolicy].
+  ///
+  /// See also:
+  ///
+  ///  * [FocusTraversalPolicy] for the API used to impose traversal order
+  ///    policy.
+  ///  * [WidgetOrderFocusTraversalPolicy] for a traversal policy that traverses
+  ///    nodes in the order they are added to the widget tree.
+  ///  * [ReadingOrderTraversalPolicy] for a traversal policy that traverses
+  ///    nodes in the reading order defined in the widget tree, and then top to
+  ///    bottom.
+  final FocusTraversalPolicy policy;
+
+  /// Returns the [DefaultFocusTraversal] that most tightly encloses the given
+  /// [BuildContext].
+  ///
+  /// The [context] argument must not be null.
+  static FocusTraversalPolicy of(BuildContext context, {bool ifAny = false}) {
+    if (context == null) {
+      return null;
+    }
+    final DefaultFocusTraversal inherited = context.inheritFromWidgetOfExactType(DefaultFocusTraversal);
+    assert(() {
+      if (ifAny) {
+        return true;
+      }
+      if (inherited == null) {
+        throw FlutterError('Unable to find a DefaultFocusTraversal widget in the context.\n'
+            'DefaultFocusTraversal.of() was called with a context that does not contain a '
+            'DefaultFocusTraversal.\n'
+            'No DefaultFocusTraversal ancestor could be found starting from the context that was '
+            'passed to DefaultFocusTraversal.of(). This can happen because there is not a '
+            'WidgetsApp or MaterialApp widget (those widgets introduce a DefaultFocusTraversal), '
+            'or it can happen if the context comes from a widget above those widgets.\n'
+            'The context used was:\n'
+            '  $context');
+      }
+      return true;
+    }());
+    return inherited?.policy;
+  }
+
+  @override
+  bool updateShouldNotify(InheritedWidget oldWidget) {
```

Comment:
This should be:
```
bool updateShouldNotify(DefaultFocusTraversal oldWidget) => policy != oldWidget.policy;
```
That's because there's no guarantee that rebuilding this widget will rebuild its descendants.

Reply:
Thanks.  Fixed.